src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SequencePoints.cs

```diff
@@ -204,26 +204,18 @@ internal BoundExpression AddConditionSequencePoint(BoundExpression condition, Bo
             // EnC source mapping only operates on statements.
             var local = _factory.SynthesizedLocal(condition.Type, containingStatement.Syntax, kind: SynthesizedLocalKind.ConditionalBranchDiscriminator);
 
-            var condConst = condition.ConstantValue;
-            if (condConst == null)
-            {
-                return new BoundSequence(
-                    condition.Syntax,
-                    ImmutableArray.Create(local),
-                    ImmutableArray.Create<BoundExpression>(_factory.AssignmentExpression(_factory.Local(local), condition)),
-                    new BoundSequencePointExpression(syntax: null, expression: _factory.Local(local), type: condition.Type),
-                    condition.Type);
-            }
-            else
-            {
-                // const expression must stay a const to not invalidate results of control flow analysis
-                return new BoundSequence(
-                    condition.Syntax,
-                    ImmutableArray.Create(local),
-                    ImmutableArray.Create<BoundExpression>(_factory.AssignmentExpression(_factory.Local(local), condition)),
-                    condition,
-                    condition.Type);
-            }
+            // Add hidden sequence point unless the condition is a constant expression.
+            // Constant expression must stay a const to not invalidate results of control flow analysis.
+            var valueExpression = (condition.ConstantValue == null) ?
+                new BoundSequencePointExpression(syntax: null, expression: _factory.Local(local), type: condition.Type) :
+                condition;
+
+            return new BoundSequence(
+                condition.Syntax,
+                ImmutableArray.Create(local),
+                ImmutableArray.Create<BoundExpression>(_factory.AssignmentExpression(_factory.Local(local), condition)),
```

Comment:
@AlekseyTs I agree it's probably not needed and I considered removing it. However i opted for consistency with C# (keeping it) for now. There are other cases that could be optimized. We can revisit post RTM.